flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -619,12 +630,17 @@ public void alterPartition(ObjectPath tablePath, CatalogPartitionSpec partitionS
 
   @Override
   public List<String> listFunctions(String dbName) throws CatalogException {
-    return Collections.emptyList();
+    return Lists.newArrayList(partitionFunctions.keySet());
```

Comment:
I don't think it is a good idea to make the functions accessible from any database. Instead, I think you should choose where you want to expose these functions.

Reply:
This interface will show all the functions under the current iceberg catalog. My idea is that these functions are registered under the iceberg catalog level, and all databases under the current iceberg catalog can be used. If the user uses a non-iceberg catalog, these functions are not visible.   I think the access boundary of these functions is the catalog level, so is it possible to not need to display or limit the database?

Reply:
I'm not sure whether Flink allows you to omit the database name or not, but that's up to Flink. Iceberg should not expose these functions as though they exist in all databases, though.

Reply:
What you said is more reasonable. It is more appropriate for this interface to display the functions registered by the user.